db/corruption_test.cc

```diff
@@ -560,6 +561,54 @@ TEST_F(CorruptionTest, FileSystemStateCorrupted) {
   }
 }
 
+TEST_F(CorruptionTest, ParaniodFileChecksOnFlush) {
+  Options options;
+  options.paranoid_file_checks = true;
+  options.create_if_missing = true;
+  Status s;
+  for (int mode = 1; mode >= -1; mode--) {
+    delete db_;
+    s = DestroyDB(dbname_, options);
+    std::shared_ptr<mock::MockTableFactory> mock =
+        std::make_shared<mock::MockTableFactory>();
+    options.table_factory = mock;
+    mock->SetCorruptionMode(mode);
+    ASSERT_OK(DB::Open(options, dbname_, &db_));
+    Build(10);
+    s = db_->Flush(FlushOptions());
+    if (mode == 0) {
+      ASSERT_OK(s);
+    } else {
+      ASSERT_NOK(s);
+    }
+  }
+}
+
+TEST_F(CorruptionTest, ParaniodFileChecksOnCompact) {
+  Options options;
+  options.paranoid_file_checks = true;
+  options.create_if_missing = true;
+  Status s;
+  for (int mode = 1; mode >= -1; mode--) {
+    delete db_;
+    s = DestroyDB(dbname_, options);
+    std::shared_ptr<mock::MockTableFactory> mock =
+        std::make_shared<mock::MockTableFactory>();
+    options.table_factory = mock;
+    ASSERT_OK(DB::Open(options, dbname_, &db_));
+    Build(10, 2);
+    mock->SetCorruptionMode(mode);
+    DBImpl* dbi = static_cast_with_check<DBImpl>(db_);
+    dbi->TEST_FlushMemTable();
```

Comment:
Maybe I missed something but how do we make sure flush doesn't hit the corruption but the compaction range below hit it?

Reply:
Re-arranged the test to set the corruption after the flush but before the compaction.